trl/trainer/grpo_trainer.py

```diff
@@ -536,6 +537,15 @@ def _prepare_inputs(self, inputs: dict[str, Union[torch.Tensor, Any]]) -> dict[s
         self._metrics["reward"].append(self.accelerator.gather_for_metrics(rewards).mean().item())
         self._metrics["reward_std"].append(self.accelerator.gather_for_metrics(std_grouped_rewards).mean().item())
 
+        if (
+            self.accelerator.is_main_process
+            and self.log_completions
+            and self.state.global_step % self.args.logging_steps == 0
+        ):
+            for prompt, completion, reward in zip(prompts, completions, rewards):
+                text = f"---\n\033[31m{prompt}\033[34m{completion}\033[0m\n--- \033[32mReward: {reward}\033[0m ---"
+                self.accelerator.print(text)
```

Comment:
Does this get logged to WandB? What I had in mind was something like Costa did for PPO here: https://github.com/huggingface/trl/blob/af4ad47035529164799be10f3fe558ee642a9880/trl/trainer/ppo_trainer.py#L733

Since GRPO produces N completions per prompt, a variant would be to explode the DataFrame column-wise, so that one has columns like ``(prompt, completion_0, reward_0, completion_1, reward_1, ..., completion_{N-1}, reward_{N-1}`

WDYT?

Reply:
Working with W&B tables is frustrating. They often don’t update properly (see [this](https://github.com/wandb/wandb/issues/2981)), generate excessive artifacts, and make it unclear whether step visualization is possible. The only somewhat reliable approach seems to be logging the entire table once at the end of training, as shown [here](https://docs.wandb.ai/tutorials/tables/#2-run-training-and-log-test-predictions), but that’s not very useful in practice. And wandb Provide very little support. That's why I avoid it usually.

I'll push this solution, you'll tell me what you think.